phoenix-core/src/it/java/org/apache/phoenix/end2end/ExplainPlanWithStatsEnabledIT.java

```diff
@@ -47,46 +53,116 @@
 import org.apache.phoenix.util.EnvironmentEdge;
 import org.apache.phoenix.util.EnvironmentEdgeManager;
 import org.apache.phoenix.util.PhoenixRuntime;
-import org.junit.BeforeClass;
+import org.apache.phoenix.util.PropertiesUtil;
+import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
 
+import com.google.common.base.Optional;
 import com.google.common.collect.Lists;
 
 /**
  * This class has tests for asserting the bytes and rows information exposed in the explain plan
  * when statistics are enabled.
  */
+@RunWith(Parameterized.class)
```

Comment:
It's too expensive to parameterize this test suite to run with different stats cache instances.     This test suite has already taking long time to run, so it's a very bad idea to double its running time. Also this test suit contains too many test cases, we don't need to run all test cases to prove JVM based cache works. My suggestion is that you can create a new sub-class of ParallelStatsEnabledIT and only create one E2E test case which is enough.